styx-scheduler-service/src/main/java/com/spotify/styx/api/SchedulerResource.java

```diff
@@ -278,8 +279,12 @@ private StatusType eventInjectorHelper(Event event) {
           || cause instanceof IllegalArgumentException) {
         // TODO: propagate error information using a more specific exception type
         return Response.forStatus(CONFLICT.withReasonPhrase(cause.getMessage()));
+      } else if (cause instanceof AlreadyInitializedException) {
```

Comment:
It better to use `findCause` in `ExceptionUtil`. The whole exception handling part would be more robust.